lambdas/services/create_document_reference_service.py

```diff
@@ -11,108 +11,136 @@
 from utils.exceptions import InvalidResourceIdException
 from utils.lambda_exceptions import CreateDocumentRefException
 from utils.lloyd_george_validator import LGInvalidFilesException, validate_lg_files
-from utils.utilities import create_reference_id, validate_id
+from utils.utilities import create_reference_id, validate_nhs_number
+
+FAILED_CREATE_REFERENCE_MESSAGE = "Create document reference failed"
+PROVIDED_DOCUMENT_SUPPORTED_MESSAGE = "Provided document is supported"
+UPLOAD_REFERENCE_SUCCESS_MESSAGE = "Upload reference creation was successful"
+UPLOAD_REFERENCE_FAILED_MESSAGE = "Upload reference creation was unsuccessful"
+PRESIGNED_URL_ERROR_MESSAGE = (
+    "An error occurred when creating pre-signed url for document reference"
+)
 
 logger = LoggingService(__name__)
 
 
 class CreateDocumentReferenceService:
-    def __init__(self, nhs_number):
-        self.nhs_number = nhs_number
+    def __init__(self):
         self.s3_service = S3Service()
         self.dynamo_service = DynamoDBService()
 
-        self.lg_s3_bucket_name = os.getenv("LLOYD_GEORGE_BUCKET_NAME")
         self.lg_dynamo_table = os.getenv("LLOYD_GEORGE_DYNAMODB_NAME")
-        self.arf_s3_bucket_name = os.getenv("DOCUMENT_STORE_BUCKET_NAME")
         self.arf_dynamo_table = os.getenv("DOCUMENT_STORE_DYNAMODB_NAME")
+        self.staging_bucket_name = os.getenv("STAGING_STORE_BUCKET_NAME")
+        self.upload_sub_folder = "user_upload"
+
+    def create_document_reference_request(
+        self, nhs_number: str, documents_list: list[dict]
+    ):
+        arf_documents: list[NHSDocumentReference] = []
+        arf_documents_dict_format: list = []
+        lg_documents: list[NHSDocumentReference] = []
+        lg_documents_dict_format: list = []
+        url_responses = {}
 
-        self.arf_documents: list[NHSDocumentReference] = []
-        self.arf_documents_dict_format: list = []
-        self.lg_documents: list[NHSDocumentReference] = []
-        self.lg_documents_dict_format: list = []
-        self.url_responses = {}
-
-    def create_document_reference_request(self, documents_list: list[dict]):
         try:
-            validate_id(self.nhs_number)
+            validate_nhs_number(nhs_number)
             for document in documents_list:
-                document_reference = self.prepare_doc_object(document)
-                self.prepare_pre_signed_url(document_reference)
-            if self.lg_documents:
-                validate_lg_files(self.lg_documents, self.nhs_number)
+                (document_reference, doc_type) = self.prepare_doc_object(
+                    nhs_number, document
+                )
+                self.check_valid_doc_type(doc_type)
```

Comment:
Is this method necessary as we're checking the types again on lines 54-58?

Reply:
I'm going with a yes, it's a fail fast condition and reusable logic.

Reply:
I was going to say yes but I've now extracted it into the default section of a switch statement